### DISCUSSION TITLE
QskLinear‘question

### DISCUSSION THREAD

**Discussion by hestrro** [uwerat/qskinny#473] [Q&A] [answered]
What should I do if I want to have QskLinearBox's additem and removeitem functions to manage objects, but I also want to manage the position and size of objects only through setGeometry?

**Comment by uwerat** [comment, marked as answer]
Every QObject manages the lifetime of its children - QskLinearBox is not special in this regard.

**Comment by hestrro** [comment]
I want to use the additem and removeitem functions that you have written. I just remembered that I can add each item by setPlacementPolicy(QskPlacementPolicy::Ignore);

**Comment by uwerat** [comment]
There is no point in adding items to a layout, when you want to set the geometries manually.

**Comment by hestrro** [comment]
![image](https://github.com/user-attachments/assets/6fd9f6f3-5418-49ef-af48-0e90d6a6b2a1)
When I want to add some graphic objects to qskscrollarea, I need a container to load these graphic objects, because here qskscrollarea can only set a sliding content

**Comment by uwerat** [comment]
Sure, but that has nothing to do with using layouts or not. You can use any QskControl as container.

**Comment by hestrro** [comment]
Yes, but I saw that the additem and removeitem functions are written in qsklineabox, so I want to use this control to implement the function.

**Comment by uwerat** [comment]
Pass the container as parent for the children - that's all you need.
Note, that overloading QskControl::updateLayout is where you should calculate and assign the geometries of the children.

**Comment by uwerat** [comment]
Note that Qt/Quick has 2 trees:

- object tree ( QObject::setParent() )
- item tree ( QQuickItem::setParentItem() )

The object tree handles the ownership, while the item tree is about geometries.

Usually the parentItem is also the parent, but you you don't need to do so.. This is different to Qt/Widgets, where the parent widget is always the parent in the object tree,
